### PR TITLE
Print to less if output is bigger than terminal window height.

### PR DESCRIPTION
**Pull request by Hiers**
Instead of having a default of 4 results, the default now is printing all 10 results and piping them into `less` if the output is bigger than the terminal window height.

This won't compile on windows because the `pager` crate is linux only, but the current binary already doesn't work well with it so I don't think it's a great loss.

### CODE REVIEW

**Comment by JojiiOfficial** [comment]
I don't like removing windows support completely. I haven't tested it on windows but if you'd make it an optional feature or something its fine

**Comment by Hiers** [comment]
I was having a lot of trouble with getting the `pager` crate to work with interactive mode. For some reason, it makes `jisho-cli` the child process and `less` the parent. ¯\_ (ツ)_/¯

In the end I ended up implementing that functionality myself, with the added benefit that it also works on windows now.

File: src/main.rs [hunk 1]
```diff
@@ -33,6 +36,17 @@ impl Default for Options {
 }
 
 fn main() -> Result<(), ureq::Error> {
+    let term_size;
+
+    if atty::is(Stream::Stdout) {
+        match terminal_size() {
+            Ok(s) => term_size = s,
+            Err(_e) => term_size = 0
+        }
```
Comment (JojiiOfficial) [inline src/main.rs]:
```rust
term_size = terminal_size().unwrap_or(0);
```

File: src/main.rs [hunk 2]
```diff
@@ -91,15 +108,48 @@ fn main() -> Result<(), ureq::Error> {
...
+
+                                    // We don't care about the return value, only whether wait failed or not
+                                    if process.wait().is_err() {
+                                        panic!("wait() was called on non-existent child process\
+                                         - this should not be possible");
+                                    }
+                                }
+                                // less not found in PATH; print normally
+                                Err(_e) => print!("{}", output)
+                            };
```
Comment (JojiiOfficial) [inline src/main.rs]: Personally I would create a separate function for this. Also please don't write such big matches as they're hard to read and the level of indentation increases by a lot. Put the Command::new()...spawn() in a variable and use that as match condition.

File: src/main.rs [hunk 3]
```diff
@@ -119,7 +169,8 @@ fn main() -> Result<(), ureq::Error> {
     Ok(())
 }
 
-fn print_item(query: &str, value: &Value) -> Option<()> {
+fn print_item(query: &str, value: &Value, output: &mut String) -> Option<usize> {
+    let mut aux;
```
Comment (JojiiOfficial) [inline src/main.rs]: Why declaring aux without value instead of simply using `let aux = ` in line 183?

**Review by JojiiOfficial — COMMENTED** [review]
Some code improvements I found left in the code. Rest looks good to me.

File: src/main.rs (hunk 2 above)
Comment (Hiers) [reply src/main.rs]: Yeah sorry, I realised this was janky only after I pushed the commit.

**Comment by JojiiOfficial** [comment]
Thanks!

**Pull request merged**